docs/reference/query-languages/esql/esql-lookup-join.md

```diff
@@ -156,9 +156,23 @@ To obtain a join key with a compatible type, use a [conversion function](/refere
 
 For a complete list of supported data types and their internal representations, see the [Supported Field Types documentation](/reference/query-languages/esql/limitations.md#_supported_types).
 
+## Implementation details
+
+This section covers key implementation aspects of `LOOKUP JOIN` that affect query behavior and results. Review these details to ensure your queries work as expected and to troubleshoot unexpected results.
+
+### Handling name collisions
+
+In case of name collisions, the newly created columns will override existing columns.
```

Comment:
@alex-spies @costin this is pre-existing info unchanged in this PR, but was wondering if there was additional guidance about workarounds we could give about collisions?

Reply:
Sure, good idea!

Like this?
```
Such columns can be preserved by using `RENAME` to assign non-conflicting names before the `LOOKUP JOIN`.
```

And yes, we realize this is clunky and we want to have a more ergonomic approach to avoid this renaming dance.